ddtrace/propagation/http.py

```diff
@@ -363,7 +363,8 @@ def _extract(headers):
         if not meta:
             meta = {}
 
-        if not meta.get(SAMPLING_DECISION_TRACE_TAG_KEY):
+        # Only set sampling decision tag if there's no head sampling decision (no sampling_priority header)
+        if not meta.get(SAMPLING_DECISION_TRACE_TAG_KEY) and sampling_priority is None:
```

Comment:
Can we update b3 and trace context headers to have the same logic. Or maybe we can put this in more central place like `HTTPPropagator.extract(...)`